inst/include/dust/random/normal.hpp

```diff
@@ -5,44 +5,46 @@
 
 #include "dust/random/generator.hpp"
 #include "dust/random/numeric.hpp"
+#include "dust/random/normal_box_muller.hpp"
+#include "dust/random/normal_ziggurat.hpp"
 
 namespace dust {
 namespace random {
 
-namespace {
+namespace algorithm {
+enum class normal {box_muller, ziggurat};
+}
 
 __nv_exec_check_disable__
-template <typename real_type, typename rng_state_type>
+template <typename real_type,
+          algorithm::normal algorithm = algorithm::normal::box_muller,
+          typename rng_state_type>
 __host__ __device__
-real_type box_muller(rng_state_type& rng_state) {
-  // This function implements the Box-Muller transform:
-  // https://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform#Basic_form
-  // Do not send a really small number to log().
-  const real_type epsilon = utils::epsilon<real_type>();
-  const real_type two_pi = 2 * M_PI;
-
-  real_type u1, u2;
-  do {
-    u1 = random_real<real_type>(rng_state);
-    u2 = random_real<real_type>(rng_state);
-  } while (u1 <= epsilon);
-
-  SYNCWARP
-  return std::sqrt(-2 * std::log(u1)) * std::cos(two_pi * u2);
-}
-
+real_type random_normal(rng_state_type& rng_state) {
```

Comment:
Is the algorithm chosen at run time or when compiling the object? Wondering whether we could just template rather than using this function

Reply:
that is a template - am I missing something?

Reply:
After discussion: it *feels* like one could overload this as we know the algorithm is known at compile time but C++ doesn't let us do that